src/auth/manager.py

```diff
@@ -0,0 +1,151 @@
+from passlib.context import CryptContext
+from jose import jwt
+from datetime import datetime
+from src.config import settings
+from fastapi import Depends, HTTPException, status
+from jose.exceptions import JWTError
+from fastapi.security import OAuth2PasswordBearer
+from src.db.conection import get_async_session
+from src.db.models import User
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+
+pwd_context = CryptContext(schemes=['bcrypt'], deprecated='auto')
+oauth_scheme = OAuth2PasswordBearer(tokenUrl='/token')
+
+def create_access_jwt(data: dict[str,str]) -> str:
+    """
+    Create an access token with the given data.
+    
+    Args:
+        data: The data to be encoded in the token
+        
+    Returns:
+        The encoded token
+    """
+    
+    data['exp'] = datetime.utcnow() + settings.JWT_ACCESS_EXP
+    data['mode'] = 'access_token'
+    return jwt.encode(data, settings.SECRET, settings.ALGORITHM)
+
+
+def create_refresh_jwt(data: dict[str,str]) -> str:
+    """
+    Create a refresh token with the given data.
+    
+    Args:
+        data: The data to be encoded in the token
+        
+    Returns:
+        The encoded token
+    """
+    
+    data['exp'] = datetime.utcnow() + settings.JWT_REFRESH_EXP
+    data['mode'] = 'refresh_token'
+    return jwt.encode(data, settings.SECRET, settings.ALGORITHM)
+
+async def get_token_data(token: str = Depends(oauth_scheme)) -> dict[str,str]:
+    """
+    Get the data from the given token.
+    
+    Args:
+        token: The token to be decoded
+    
+    Returns:
+        The decoded token data
+    """
+    
+    try:
+        data = jwt.decode(token, settings.SECRET, settings.ALGORITHM)
+        data['token'] = token
+    except JWTError:
+        raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail='invalid authorization credentials'
+            )
+
+    
+    if 'user_email' not in data and 'mode' not in data:
+        raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail='invalid authorization credentials'
+            )
+    
+    return data
+
+
+async def refresh_token(data: str = Depends(get_token_data), session: AsyncSession = Depends(get_async_session)) -> dict[str,str]:
+    """
+    Refresh the given token.
+    
+    Args:
+        token: The token to be refreshed
+        session: The database session
+        
+    Returns:
+        A dictionary with the new access,refresh tokens and the token type
+    """
+    
+    if data['mode'] != 'refresh_token':
+        raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail='not a refresh token'
```

Comment:
```suggestion
                detail='Not a refresh token'
```
Does it metter for front if they will wait for messages starting with capital letter?